### PR TITLE
Add release documentation

### PR DESCRIPTION
**Pull request by JakeGinnivan**
@gep13 @asbjornu @Jericho 

When this release process was created, a few of the manual steps couldn't be automated. There may be ways to improve

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -68,3 +68,27 @@ The last line is the most important. `AssertFullSemver` will run GitVersion and
...
+1) All the artifacts should upload nicely
+![docs/img/release-3-deploy.png](docs/img/release-3-deploy.png)
+1) Head over to GitHub releases, you should have a draft release, download a copy of the release notes
+![docs/img/release-4-deploy.png](docs/img/release-4-deploy.png)
+1) Edit the release and do the following:
+    1. Remove the build metadata from the tag and title (the + and everything after it)
+    2. Paste the downloaded release notes in, you can clean them up if you want otherwise there may be closed issues which were questions etc. 
+    3. Tick the pre-release box if it's pre-release
+    4. Press Publish
+1) Publishing tags (a git tag) the release commit, this will trigger another appveyor build which only builds tags, this build uses deploy.cake. It downloads the artifacts from that GitHub release, then performs the release
```
Comment (pascalberger) [inline CONTRIBUTING.md]: I don't really understand this point. Isn't the tag automatically created through publishing the GitHub release (point before)? Are there additional tags which needs to be created?

Comment (JakeGinnivan) [reply CONTRIBUTING.md]: Will clarify in the docs.

We push a *non published* github release, so it doesn't create a tag. The tag is created when you publish the release from the GitHub site

Comment (JakeGinnivan) [reply CONTRIBUTING.md]: Sorry, AppVeyor creates a draft, I wanted to call out that publish at this point creates the tag and that runs the build

**Comment by asbjornu** [comment]
Can't we auto-publish releases from AppVeyor without any manual steps? Builds without a tag can get a pre-release label and builds with a tag are considered stable releases. Wouldn't that work?

**Comment by JakeGinnivan** [comment]
We can auto publish, but I don't know how to automate the release notes part of it..

Also the release notes require manual intervention most of the time :( 

Happy for us to change what is setup, this is just documenting the current setup

**Comment by asbjornu** [comment]
Can't we use GitReleaseManager or similar to at least sketch out release notes so they can be manually edited on GitHub? I've never gotten it to work in my repos, but I assume someone must know how to get it to work?

**Comment by JakeGinnivan** [comment]
Not with appveyor deployments afaik, we can't run tools in that step.

We could put it in the actual deployment steps though

**Comment by asbjornu** [comment]
We could instrument that through Cake, no?

**Comment by JakeGinnivan** [comment]
Yeah probably, the release notes are probably not good enough quality to just be used, but can edit after the fact if so. Happy for it to be updated as I said :)

**Comment by asbjornu** [comment]
Is the release note part of this something you can help add, @gep13? It would be so nice to get this and a more automated build pipeline in so we can release this more consistently. I'm a bit ashamed of the state of the project now, tbh. 😞 

What I would like is an automatic beta-release for every merge to `master` and then a stable release with automatic release-creation on Github if the built commit is tagged. Just getting beta packages out the door frequently and consistently would be awesome.

**Comment by JakeGinnivan** [comment]
We could just go back to the drawing board with the release process. Completely up to you guys :)

**Comment by ovebastiansen** [comment]
Any chance you also can upload beta versions as preview version into vsts extensions? Right now its a bit hard to use the beta version in vsts.

**Comment by asbjornu** [comment]
@ovebastiansen: If we redo the release process a bit, like I propose above (https://github.com/GitTools/GitVersion/pull/1338#issuecomment-350395507), beta-releases should be published to all possible package sources every time we merge to `master`. If you can help out with that, it'd be great!

**Comment by ovebastiansen** [comment]
@asbjornu, sry there has been way to much stuff happening lately.
How stuck are you on the current setup? I am mostly familiar with vsts (have not used cake yet) and this writeup seems promising https://oren.codes/2018/05/15/oss-build-and-release-with-vsts/
The build system can still use github for code and build whenever any thing is checked into here, but adding the Release parts in vsts might get us somewhere we want to be, regarding automating stuff.

**Pull request merged**